gordo_components/builder/build_model.py

```diff
@@ -89,30 +94,66 @@ def build_model(
         # Cross validate
         logger.debug("Starting cross validation")
         start = time.time()
-
-        scores: Dict[str, Any]
-
+        scores: Dict[str, Any] = dict()
         if hasattr(model, "predict"):
-
-            cv_scores = cross_val_score(
+            metrics_list = [
+                explained_variance_score,
+                r2_score,
+                mean_squared_error,
+                mean_absolute_error,
+            ]
+            metrics_dict = {}
+            for metric in metrics_list:
+                for index, col in enumerate(y.columns):
+
+                    def my_custom_score(y_true, y_pred, metric=metric, index=index):
```

Comment:
And this thing, can it be moved out, and given a better name? And maybe even a docstring. But it can be hidden (starting with a underscore)

Reply:
I tried to remove this function out of the for loops and create a wrapper, something like this:

```            def my_custom_score(y_true, y_pred, metric=r2_score, index=0):
                score = metric(y_true.values[:, index], y_pred[:, index])
                return score
```
and then in the for loop use a lambda function:
```    for metric in metrics_list:
        for index, col in enumerate(y.columns):
            metric_str = metric.__name__
            metrics_dict.update(
                {
                    metric_str.replace("_", "-")
                    + f'-{col.replace(" ", "-")}': make_scorer(
                        metric_wrapper(lambda y_true,y_pred: my_custom_score(y_true,y_pred,metric=metric,index=index)
                    )
                }
            )
```
but im getting the same results for different metrics/indexes...it looks like (since the lambda function depends only on y_true and y_pred that the metric and index are not being iterated...